lib/auth/integration/integrationv1/awsra.go

```diff
@@ -87,3 +91,136 @@ func (s *Service) generateAWSRACredentialsWithoutAuthZ(ctx context.Context, req
+		return nil, trace.Wrap(err)
+	}
+	spec := integration.GetAWSRolesAnywhereIntegrationSpec()
+	if spec == nil {
+		return nil, trace.BadParameter("integration %q is not an AWSRA integration", integrationName)
+	}
+
+	return spec, nil
+}
+
+// AWSRolesAnywhereServiceConfig holds configuration options for the AWSRolesAnywhere Integration gRPC service.
+type AWSRolesAnywhereServiceConfig struct {
+	// IntegrationService is the service that provides access to integration credentials.
+	IntegrationService *Service
+	// Authorizer is the authorizer used to check access to the integration.
+	Authorizer authz.Authorizer
+
+	Clock  clockwork.Clock
+	Logger *slog.Logger
+}
+
+// CheckAndSetDefaults checks the AWSRolesAnywhereServiceConfig fields and returns an error if a required param is not provided.
+// Authorizer and IntegrationService are required params.
+func (s *AWSRolesAnywhereServiceConfig) CheckAndSetDefaults() error {
+	if s.Authorizer == nil {
+		return trace.BadParameter("authorizer is required")
+	}
+
+	if s.IntegrationService == nil {
+		return trace.BadParameter("integration service is required")
+	}
+
+	if s.Clock == nil {
+		s.Clock = clockwork.NewRealClock()
+	}
+
+	if s.Logger == nil {
+		s.Logger = slog.With(teleport.ComponentKey, "integrations.awsra.service")
+	}
+
+	return nil
+}
+
+// AWSRolesAnywhereService implements the teleport.integration.v1.AWSRolesAnywhereService RPC service.
+type AWSRolesAnywhereService struct {
+	integrationpb.UnimplementedAWSRolesAnywhereServiceServer
+
+	integrationService *Service
+	authorizer         authz.Authorizer
+	logger             *slog.Logger
+	clock              clockwork.Clock
+}
+
+// NewAWSRolesAnywhereService returns a new AWSRolesAnywhereService.
+func NewAWSRolesAnywhereService(cfg *AWSRolesAnywhereServiceConfig) (*AWSRolesAnywhereService, error) {
+	if err := cfg.CheckAndSetDefaults(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return &AWSRolesAnywhereService{
+		integrationService: cfg.IntegrationService,
+		logger:             cfg.Logger,
+		authorizer:         cfg.Authorizer,
+		clock:              cfg.Clock,
+	}, nil
+}
+
+var _ integrationpb.AWSRolesAnywhereServiceServer = (*AWSRolesAnywhereService)(nil)
+
+// ListRolesAnywhereProfiles returns a paginated list of Roles Anywhere Profiles.
+func (s *AWSRolesAnywhereService) ListRolesAnywhereProfiles(ctx context.Context, req *integrationpb.ListRolesAnywhereProfilesRequest) (*integrationpb.ListRolesAnywhereProfilesResponse, error) {
+	authCtx, err := s.authorizer.Authorize(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	if err := authCtx.CheckAccessToKind(types.KindIntegration, types.VerbUse); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	// ListRolesAnywhereProfiles uses the ProfileSync config's Profile and Role.
+	spec, err := s.integrationService.getAWSRolesAnywhereIntegrationSpec(ctx, req.GetIntegration())
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	trustAnchor := spec.TrustAnchorARN
```

Comment:
Do we even need to to this here? It looks like `generateAWSRACredentialsWithoutAuthZ` will look up the trustAnchor if not provided.